include/z64transition.h

```diff
@@ -164,30 +165,28 @@ typedef enum {
     /* 86 */ TRANS_TYPE_86 = 86
 } TransitionType;
 
-typedef enum {
-    /* 0 */ FBDEMO_FADE,
-    /* 1 */ FBDEMO_TRIFORCE,
-    /* 2 */ FBDEMO_WIPE1,
-    /* 3 */ FBDEMO_WIPE3,
-    /* 4 */ FBDEMO_WIPE4,
-    /* 5 */ FBDEMO_CIRCLE,
-    /* 6 */ FBDEMO_WIPE5
+#define DEFINE_TRANSITION(enumValue, _structName, _instanceName, _name) enumValue,
+#define DEFINE_TRANSITION_INTERNAL(enumValue, _structName, _instanceName) enumValue,
+
+typedef enum FbDemoType {
+    #include "tables/transition_table.h"
+    /* 7 */ FBDEMO_MAX
 } FbDemoType;
 
+#undef DEFINE_TRANSITION
+#undef DEFINE_TRANSITION_INTERNAL
+
 #define TRANS_NEXT_TYPE_DEFAULT 0xFF
 
+#define DEFINE_TRANSITION(_enumValue, structName, instanceName, _name) structName instanceName;
+#define DEFINE_TRANSITION_INTERNAL(_enumValue, structName, instanceName) structName instanceName;
+
 typedef struct {
     /* 0x000 */ s16 transitionType;
     /* 0x002 */ s8 fbdemoType;
     /* 0x003 */ char unk_003[0x5];
     /* 0x008 */ union {
-        TransitionFade fade;
-        TransitionCircle circle;
-        TransitionTriforce triforce;
-        TransitionWipe1 wipe1;
-        TransitionWipe3 wipe3;
-        TransitionWipe4 wipe4;
-        TransitionWipe5 wipe5;
+        #include "tables/transition_table.h"
     } instanceData;
```

Comment:
Hmm, thoughts on maybe creating a union typedef instead of the include in the middle? Just a suggestion, as I'm not 100% sure myself.

Something like this
```
#define DEFINE_TRANSITION(_enumValue, structName, instanceName, _name) structName instanceName;
#define DEFINE_TRANSITION_INTERNAL(_enumValue, structName, instanceName) structName instanceName;

typedef union {
    #include "tables/transition_table.h"
} TransitionInstance;

#undef DEFINE_TRANSITION
#undef DEFINE_TRANSITION_INTERNAL

typedef struct {
    /* 0x000 */ s16 transitionType;
    /* 0x002 */ s8 fbdemoType;
    /* 0x003 */ char unk_003[0x5];
    /* 0x008 */ TransitionInstance transition;
    /* 0x220 */ char unk_220[0x10];
    /* 0x230 */ void* (*init)(void* transition);
    /* 0x234 */ void  (*destroy)(void* transition);
    /* 0x238 */ void  (*update)(void* transition, s32 updateRate);
    /* 0x23C */ void  (*draw)(void* transition, Gfx** gfxP);
    /* 0x240 */ void  (*start)(void* transition);
    /* 0x244 */ void  (*setType)(void* transition, s32 type);
    /* 0x248 */ void  (*setColor)(void* transition, u32 color);
    /* 0x24C */ void  (*setEnvColor)(void* transition, u32 color);
    /* 0x250 */ s32   (*isDone)(void* transition);
    /* 0x254 */ char unk_254[0x4];
} TransitionContext; // size = 0x258
```

Reply:
It feels cleaner yeah, I'll go with this instead

Reply:
I'm lost as to why that define is there to begin with? C lets you put defines in structs?? Is this required to match?

Reply:
The define is there to keep the union that the PlayState uses to store the actual instance in sync with the transition table. 
I also didn't really like the define in the middle of the struct, which is why I suggested a non anonymous union, and use the defines with that, than use the union normally.